DOC-11538 Release note for 7.2.4 - work in progress.

RayOffiah opened the pull request.

RayOffiah committed:
Added a features section to separate enhancements from bug fixes.
Corrected a mislabeled link.
Changed the release date.
Removed the [backport] entries from the release note.
Work in progress.
DOC-11538 Release note for 7.2.4 - initial commit

RichardSmedley commented on modules/release-notes/partials/docs-server-7.2.4-release-note.adoc:
```diff
@@ -0,0 +1,130 @@
+
+
+[#release-724]
+== Release 7.2.4 (November 2023)
```
Should this be 7.2.3?

RayOffiah commented:
Hi, yes. Good question. 🙂

7.2.3 is being released early next month, along with 7.1.6 

These are the fixes that were part of 7.2.3 but they've been moved to the 7.2.4.

RayOffiah committed:
Minor spacing correction.

bfavini commented on modules/release-notes/partials/docs-server-7.2.4-release-note.adoc:
```diff
@@ -0,0 +1,104 @@
+[#release-724]
+== Release 7.2.4 (January 2024)
+
+Couchbase Server 7.2.4 was released in November 2023.
```
@RayOffiah you'll need to update this--January 2024

bfavini commented on modules/release-notes/partials/docs-server-7.2.4-release-note.adoc:
```diff
@@ -0,0 +1,104 @@
+== Release 7.2.4 (January 2024)
+
+Couchbase Server 7.2.4 was released in November 2023.
+This maintenance release contains new features and fixes to issues.
```
This maintenance release contains new features and fixes.

bfavini commented on modules/release-notes/partials/docs-server-7.2.4-release-note.adoc:
```diff
@@ -0,0 +1,104 @@
+|===
+|Feature | Description
+
+| https://issues.couchbase.com/browse/MB-59503/[MB-59503]
```
@RayOffiah I'm asking @ggray-cb if he has more on this feature, which was backported from 7.6.

RayOffiah replied:
Okay, let me know.  👍🏾

bfavini commented on modules/release-notes/partials/docs-server-7.2.4-release-note.adoc:
```diff
@@ -0,0 +1,104 @@
+|Feature | Description
+
+| https://issues.couchbase.com/browse/MB-8925/[MB-58925]
+| Couchbase Server 7.2.2 currently uses Erlang{nbsp}24. As Couchbase Server 7.2 will be maintained until 2026, Erlang has been upgraded to version{nbsp}25 to ensure we receive future code fixes.
```
I would simply say:

Erlang has been upgraded to version 25 in Couchbase Server 7.2.4.

bfavini commented on modules/release-notes/partials/docs-server-7.2.4-release-note.adoc:
```diff
@@ -0,0 +1,104 @@
+
+|  https://issues.couchbase.com/browse/MB-59261/[MB-59261]
+| `Memcached` alert added. +
+Add an alert when the number of memcached connections has exceeded 90% of the limit.
```
Couchbase Server now issues an alert when the number of memcached connections exceeds 90% of the limit.

bfavini commented on modules/release-notes/partials/docs-server-7.2.4-release-note.adoc:
```diff
@@ -0,0 +1,104 @@
+|Issue | Description | Resolution
+
+| https://issues.couchbase.com/browse/MB-59237/[MB-59237]
+| Record OS disk I/O metrics into Prometheus. Currently, there is not enough information recorded in the logs to diagnose slowdowns in the system related to Disk I/O
```
@RayOffiah I've added a comment to this JIRA asking the engineer to provide a brief description of the problem and resolution. It's not clear to me whether all of this was actually implemented.

RayOffiah replied:
Okay, I'll chase that one up.

bfavini commented on modules/release-notes/partials/docs-server-7.2.4-release-note.adoc:
```diff
@@ -0,0 +1,104 @@
+|Issue | Description | Resolution
+
+| https://issues.couchbase.com/browse/MB-59646/[MB-59646]
+| Index has zero entries when running `NESTED ARRAY FLATTEN_KEYS()`.
```
@RayOffiah I've also asked for a brief description in the ticket. I think we need to polish this description. Something like:

Indexes were not correctly created on  nested arrays using the FLATTEN_KEYs function because null or missing entries were not expanded. This issue has been resolved.

RayOffiah replied:
Okay, I'll chase that up.

bfavini commented on modules/release-notes/partials/docs-server-7.2.4-release-note.adoc:
```diff
@@ -0,0 +1,104 @@
+
+
+| https://issues.couchbase.com/browse/MB-57903/[MB-57903]
+| Deep-nested correlations aren’t properly detected and use cached results
```
Deeply nested correlations were not properly detected and used cached results. This issue has been fix.

RayOffiah committed:
https://issues.couchbase.com/browse/MB-59237 is an enhancement, rather than a bug fix. Moved to the `new features` section.
Added more detail to https://issues.couchbase.com/browse/MB-59503/
Heading set to wrong level
Missing quote on table definition.
Added new entries following review.
Changes for review.
Added https://issues.couchbase.com/browse/MB-58648

bfavini commented on modules/release-notes/partials/docs-server-7.2.4-release-note.adoc:
```diff
@@ -0,0 +1,150 @@
+[#release-724]
+== Release 7.2.4 (January 2024)
+
+Couchbase Server 7.2.4 was released in November 2023. +
```
@RayOffiah please update--this should say Jan 2024, not November 2023.

bfavini commented on modules/release-notes/partials/docs-server-7.2.4-release-note.adoc:
```diff
@@ -0,0 +1,150 @@
+* Disk read latency.
+
+| https://issues.couchbase.com/browse/MB-59503/[MB-59503]
+a| The request is for `ns_server` to provide a REST interface to enter the minimum duration (in minutes) that previously rotated audit files will be retained. This value will be converted to seconds and passed through to memcached. The audit daemon, run during the log rotation task, will remove any files created more than X seconds ago. +
```
The developer added this description in the ticket:

When a query has nested correlation references, e.g., query1 contains a
subquery query2, which in turn contains another subquery query3, and
query3 is correlated with query1 (but query2 is not correlated), the
nested correlation is not properly detected and the subquery (query2)
is not properly marked as correlated.

Fix:

Correlation references are detected and marked in different levels of
subqueries, and nested correlation references are properly propagated
up such that all subqueries are properly marked when correlated.

bfavini replied:
Gary's doc on this: 

You can now have Couchbase Server prune rotated audit logs after a period of time. 
You set how long  Couchbase Server should keep audit logs by using the new `pruneAge` parameter for the `/settings/audit` endpoint. 
The default value of 0 means that Couchbase Server does not prune audit logs. 
See xref:rest-api:rest-auditing.adoc[Configure Auditing].

ggray-cb committed:
* Added edit requested by Beth.
* Added known issue from K8S-3332

ggray-cb committed:
Accidentally added .vscode file. Undoing.

ggray-cb committed:
Updating pruneAge release note with Beth's suggestion of using my What's New description.

bfavini reviewed (approved):
approved

ggray-cb committed:
Changed known issue section to be in a table like the others.
